llama.cpp

```diff
@@ -8773,9 +8978,39 @@ static void llama_set_inputs(llama_context & lctx, const llama_batch & batch) {
         ggml_backend_tensor_set(lctx.inp_pos, batch.pos, 0, n_tokens*ggml_element_size(lctx.inp_pos));
     }
 
+    {
+        GGML_ASSERT(lctx.inp_out_ids && "every model type must skip unused outputs");
```

Comment:
`inp_out_ids` is skipped when there are no outputs, so this assert is not correct. `inp_out_ids` should be set to `nullptr` in `llm_build_context::init`.

Reply:
Thanks, should be fixed now.

Reply:
This is not enough to fix it, as it is possible to have batches without outputs. For example, when evaluating a prompt larger than `n_batch`, the first batches will have no outputs. However this may not be relevant after fixing the pipeline parallelism issue.

Reply:
I understand the problem, but what's weird is that I didn't run into it before, even when I tried. It was probably because `lctx.inp_out_ids` was initialized to a garbage address and it somehow still worked.